core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java

```diff
@@ -2345,645 +3274,974 @@ public void testBinaryLiteralToHex()
+                new ShowStats(
+                        Optional.of(location(1, 1)),
+                        new Table(expectedQualifiedName2)));
 
-        for (String fullName : tableNames) {
-            QualifiedName qualifiedName = makeQualifiedName(fullName);
-            assertStatement(format("SHOW STATS FOR %s", qualifiedName), new ShowStats(new Table(qualifiedName)));
-        }
+        assertThat(statement(format("SHOW STATS FOR %s", expectedQualifiedName3))).isEqualTo(
+                new ShowStats(
+                        Optional.of(location(1, 1)),
+                        new Table(expectedQualifiedName3)));
     }
 
     @Test
     public void testShowStatsForQuery()
     {
-        String[] tableNames = {"t", "s.t", "c.s.t"};
-
-        for (String fullName : tableNames) {
-            QualifiedName qualifiedName = makeQualifiedName(fullName);
-            assertStatement(format("SHOW STATS FOR (SELECT * FROM %s)", qualifiedName),
-                    createShowStats(qualifiedName, ImmutableList.of(new AllColumns()), Optional.empty()));
-            assertStatement(format("SHOW STATS FOR (SELECT * FROM %s WHERE field > 0)", qualifiedName),
-                    createShowStats(qualifiedName,
-                            ImmutableList.of(new AllColumns()),
-                            Optional.of(
-                                    new ComparisonExpression(ComparisonExpression.Operator.GREATER_THAN,
-                                            new Identifier("field"),
-                                            new LongLiteral("0")))));
-            assertStatement(format("SHOW STATS FOR (SELECT * FROM %s WHERE field > 0 or field < 0)", qualifiedName),
-                    createShowStats(qualifiedName,
-                            ImmutableList.of(new AllColumns()),
-                            Optional.of(
-                                    new LogicalBinaryExpression(LogicalBinaryExpression.Operator.OR,
-                                            new ComparisonExpression(ComparisonExpression.Operator.GREATER_THAN,
-                                                    new Identifier("field"),
-                                                    new LongLiteral("0")),
-                                            new ComparisonExpression(ComparisonExpression.Operator.LESS_THAN,
-                                                    new Identifier("field"),
-                                                    new LongLiteral("0"))))));
-        }
-    }
+        // SELECT all cases
 
-    private static ShowStats createShowStats(QualifiedName name, List<SelectItem> selects, Optional<Expression> where)
-    {
-        return new ShowStats(
-                new TableSubquery(simpleQuery(new Select(false, selects),
-                        new Table(name),
-                        where,
-                        Optional.empty())));
+        QualifiedName qualifiedName = qualifiedName(location(1, 31), "t");
```

Comment:
Inline those. The test is easier to read with all pieces in place. If it's formatted right, the size of the case is not an issue.